pkg/repository/local_git.go

```diff
@@ -345,24 +363,104 @@ func (lg *LocalGit) GetDirContentSHA(ctx context.Context, gitPath, dirPath strin
 
 	filteredLS := strings.Join(filteredLines, "\n")
 
-	hashObjectCmdArgs := []string{"--no-optional-locks", "hash-object", "--stdin"}
-	output, err := lg.exec.RunPipeCommands(ctx, gitPath, "echo", []string{filteredLS}, lg.gitPath, hashObjectCmdArgs)
+	// As the list of files might be too long, we write it to a file and calculate the hash of the file
+	f, err := lg.fs.Create(filepath.Join(config.GetOktetoHome(), fmt.Sprintf("git-untracked-%s-%d", filepath.Base(dirPath), time.Now().UnixNano())))
+	if err != nil {
+		oktetoLog.Debugf("failed to create untracked file: %v", err)
+		return "", err
+	}
+
+	defer func() {
+		err := lg.fs.Remove(f.Name())
+		if err != nil {
+			oktetoLog.Debugf("failed to close untracked file: %v", err)
+		}
+	}()
+
+	_, err = f.WriteString(filteredLS)
+	if err != nil {
+		oktetoLog.Debugf("failed to write untracked file: %v", err)
+		return "", err
+	}
+
+	hashObjectCmdArgs := []string{"--no-optional-locks", "hash-object", f.Name()}
+	output, err := lg.exec.RunCommand(ctx, dirPath, lg.gitPath, hashObjectCmdArgs...)
 
 	if err != nil {
+		oktetoLog.Debugf("failed to calculate hash of directory %s: %v", dirPath, err)
 		return handleError(err)
 	}
 	return string(output), nil
 }
 
+// getDirContentSHAWithoutIgnore calculates the SHA of the content of the given directory using git ls-files and git hash-object
+func (lg *LocalGit) getDirContentSHAWithoutIgnore(ctx context.Context, gitPath, dirPath string, fixAttempt int) (string, error) {
```

Comment:
It is the content of the function `GitDirContent` before the change. It can be found [here](https://github.com/okteto/okteto/blob/c33b3258595857747075f99ebf6508569a8665de/pkg/repository/local_git.go#L273)